src/tree/mod.rs

```diff
@@ -413,72 +413,70 @@ impl RatchetTree {
         credential_bundle: &CredentialBundle,
         group_context: &[u8],
         new_leaves_indexes: HashSet<&LeafIndex>,
-    ) -> (UpdatePath, KeyPackageBundle) {
+    ) -> Result<(UpdatePath, KeyPackageBundle), TreeError> {
         // Generate new keypair
         let own_index = self.own_node_index();
+        let own_key_package = self.own_key_package();
 
         // Replace the init key in the current KeyPackage
-        let mut key_package_bundle =
-            KeyPackageBundle::from_rekeyed_key_package(self.own_key_package());
+        let key_package_bundle_unsigned =
+            KeyPackageBundlePayload::from_rekeyed_key_package(own_key_package);
+        // FIXME: THIS IS UNNECESSARY
```

Comment:
Is this because we sign the KPB again a few lines below? I guess this is due to a restriction that exists because of the split types for KP and KPB and would require some more refactoring to go away. Is that correct? If so, we should probably file an issue.